src/GraphCoarsenBuilder.hpp

```diff
@@ -42,10 +42,13 @@ class MBuilder
     /**
        @brief Build the assembled M for the local processor
      */
-    virtual std::unique_ptr<mfem::SparseMatrix> BuildAssembledM() const = 0;
+    std::unique_ptr<mfem::SparseMatrix> BuildAssembledM() const;
 
     /**
-       @brief Set weights on aggregates for assembly of mass matrix.
+       @brief Assemble the rescaled M for the local processor
+
+       Build the assembled M with "element" contribution inversely scaled
```

Comment:
This makes sense to me now, but maybe for others (or for me in six months) describe briefly what "element contribution" means.

Or maybe just remove these words to avoid confusion? The interface is actually just aggs; for each agg you supply an (inverse) weight, that may be all the user needs to know.

Reply:
Since `MBuilder` is a general class for any level, this function works for fine level as well. For each level, this function actually takes the element (inverse) weight in that level. In particular, an element in the coarse level is an aggregate in the fine level. 

In any case, this whole thing only makes sense to me in finite volume, so I find it hard to document it in graph language. I will take your advice to remove those words.